src/hera/workflows/_mixins.py

```diff
@@ -579,34 +579,39 @@ def _build_arguments(self) -> Optional[ModelArguments]:
         elif isinstance(self.arguments, ModelArguments):
             return self.arguments
 
+        def add_argument(k: str, v: Any, result: ModelArguments):
+            if isinstance(v, Parameter):
+                value = v.with_name(k).as_argument()
```

Comment:
nit: maybe call it `parameter` on the three param branches (currently it's `value`) and `artifact` (currently it's `copy_art`) on the other two?